### PR TITLE
Add ID attribute and parse edge information with it

### PR DESCRIPTION
**Pull request by jazhawang**
Add an ID attribute to the edge and node Dot attributes. This is so that we can retrieve this information when we parse the generated SVG. This edge information is saved in the DB and then used to generate the response from the server.

Also add some cosmetic changes.
 - Rename some functions to be more consistent with the surrounding functions. For example `performParse` -> `performParseFromFile`, which is in line with `performParseFromMemory`
 - prettify some code

### CODE REVIEW

File: app/Svg/Parser.hs [hunk 1]
```diff
@@ -38,78 +38,80 @@ import Data.List.Split (splitOn)
...
-    performParse "(unofficial) Portuguese" "prt2015.svg"
-    performParse "(unofficial) Slavic" "sla2015.svg"
-    performParse "(unofficial) East Asian Studies" "eas2015.svg"
-    performParse "(unofficial) English" "eng2015.svg"
-    performParse "(unofficial) History and Philosophy of Science" "hps2015.svg"
-    performParse "(unofficial) History" "his2015.svg"
-    performParse "(unofficial) Geography" "ggr2015.svg"
-    performParse "(unofficial) Aboriginal" "abs2015.svg"
-    performParse "(unofficial) German" "ger2015.svg"
+    mapM_ (uncurry performParseFromFile)
```
Comment (jazhawang) [inline app/Svg/Parser.hs:41]: just some style changes

File: app/Svg/Parser.hs [hunk 2]
```diff
@@ -38,78 +38,80 @@ import Data.List.Split (splitOn)
...
 parseDynamicSvg :: T.Text -> T.Text -> IO ()
 parseDynamicSvg graphName graphContents =
     runSqlite databasePath $ performParseFromMemory graphName graphContents
 
 -- | The starting point for parsing a graph with a given title and file.
-performParse :: T.Text -- ^ The title of the graph.
-             -> String -- ^ The filename of the file that will be parsed.
-             -> SqlPersistM ()
-performParse graphName inputFilename = do
+performParseFromFile :: T.Text -- ^ The title of the graph.
```
Comment (jazhawang) [inline app/Svg/Parser.hs:69]: `performParse` to `performParseFromFile` to match `performParseFromMemory`

File: app/Svg/Parser.hs (hunk 1 above)
Comment (david-yz-liu) [reply app/Svg/Parser.hs:41]: Don't mix style or refactoring changes and functionality changes in the same PR. As far as I can tell, the changes in this file from here to 113 are only style changes, and so should be pulled out into a separate PR.

Also I don't like inserting whitespace to vertically align list arguments (on the line below). Use a single space after each comma.

File: app/Svg/Parser.hs [hunk 3]
```diff
@@ -255,27 +262,28 @@ parsePathHelper key trans pathTag =
...
 -- | Create an ellipse from an open ellipse tag.
 parseEllipse :: GraphId
              -> [Tag T.Text]
              -> [Shape]
 parseEllipse key tags =
-    map (parseEllipseHelper key trans) (filter (TS.isTagOpenName "ellipse") tags)
+    map (parseEllipseHelper key trans gid) (filter (TS.isTagOpenName "ellipse") tags)
     where
         trans = getTransform $ head tags
-
```
Comment (david-yz-liu) [inline app/Svg/Parser.hs:270]: Keep this blank line.

File: app/Svg/Parser.hs [hunk 4]
```diff
@@ -255,27 +262,28 @@ parsePathHelper key trans pathTag =
...
+        gid = (fromAttrib "id" $ head tags)
 
 parseEllipseHelper :: GraphId     -- ^ The related graph id.
                    -> Point       -- ^ The translation to apply.
+                   -> T.Text
```
Comment (david-yz-liu) [inline app/Svg/Parser.hs:281]: Add a comment here for the purpose of this parameter. (The `-- ^` makes the description picked up by Haddock.)

File: app/DynamicGraphs/WriteRunDot.hs [hunk 5]
```diff
@@ -51,7 +51,7 @@ generateAndSavePrereqResponse taken courses = do
       parseDynamicSvg graphHash $ decodeUtf8 bString
       storedGraph <- getGraph graphHash
       return $ fromMaybe graphNotFound storedGraph
-  where
```
Comment (david-yz-liu) [inline app/DynamicGraphs/WriteRunDot.hs:54]: Revert this change (see comment below).

**Comment by jazhawang** [comment]
sorry for the wait! reverted the cosmetic changes

File: app/Svg/Parser.hs (hunk 4 above)
Comment (jazhawang) [reply app/Svg/Parser.hs]: done. Added small description

File: app/Svg/Parser.hs (hunk 1 above)
Comment (jazhawang) [reply app/Svg/Parser.hs]: ok i reverted it back to the original

**Review by david-yz-liu — APPROVED** [review]
